File: app/src/main/java/com/stevecrossin/whatcanicook/entities/RecipeIngredientsDao.java
package com.stevecrossin.whatcanicook.entities;



import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.ArrayList;
import java.util.List;

@Dao
public interface RecipeIngredientsDao {

    /**
     * Get all recipe ingredients from the recipeIngredients database.
     */
    @Query("SELECT * FROM recipeingredients;")
    List<RecipeIngredients> getAllRecipesAndIngredients();

    /**
     * Adds a list of recipe ingredients from an array into the recipeIngredients database
     */
    @Insert
    void addRecipeIngredients(ArrayList<RecipeIngredients> recipeIngredients);
}

File: app/src/main/java/com/stevecrossin/whatcanicook/entities/RecipeIngredientsDao.kt
package com.stevecrossin.whatcanicook.entities

import android.arch.persistence.room.Dao
import android.arch.persistence.room.Insert
import android.arch.persistence.room.Query
import java.util.*

@Dao
interface RecipeIngredientsDao {

    /**
     * Get all recipe ingredients from the recipeIngredients database.
     */
    @get:Query("SELECT * FROM recipeingredients;")
    val allRecipesAndIngredients: List<RecipeIngredients>

    /**
     * Adds a list of recipe ingredients from an array into the recipeIngredients database
     */
    @Insert
    fun addRecipeIngredients(recipeIngredients: ArrayList<RecipeIngredients>)
}
